lib/jsonapi/utils.rb

```diff
@@ -42,8 +42,7 @@ def jsonapi_render_bad_request
     end
 
     def jsonapi_render_not_found(exception)
-      setup_request
-      id = exception.message.match(/=(\d+)/)[1]
```

Comment:
@tiagopog this regex was failing in our test suite because we are using UUIDs as primary keys, which aren't strings of strictly digits.